cmd/kapacitord/main.go

```diff
@@ -88,16 +89,44 @@ func (m *Main) Run(args ...string) error {
 		}
 
 		signalCh := make(chan os.Signal, 1)
-		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
 		m.Logger.Println("I! Listening for signals")
 
 		// Block until one of the signals above is received
-		select {
-		case <-signalCh:
-			m.Logger.Println("I! Signal received, initializing clean shutdown...")
-			go func() {
-				cmd.Close()
-			}()
+	Loop:
+		for {
+			select {
```

Comment:
This select statement has only a single case, which means this can be simplified to `for s := range signalCh {...}`.

Reply:
Hmm, ok so further down we check for a second signal. Maybe we should just keep a state about the signals we have received so its clear how transitions occur.